test/lib/yarn.test.ts

```diff
@@ -81,6 +81,27 @@ const SCENARIOS_WITH_FILES = [
     workspace: 'missing-name',
     includeDev: false,
   },
+  {
+    name:
+      'Parse yarn.lock with deeply out of sync app because of local workspaces (root)',
+    workspace: 'out-of-sync-app',
+    includeDev: false,
+    strict: false,
+  },
+  {
+    name:
+      'Parse yarn.lock with deeply out of sync app because of local workspaces (app)',
+    workspace: 'out-of-sync-app/packages/app',
+    includeDev: false,
+    strict: false,
```

Comment:
let's match the named parameter to avoid ambiguity
```suggestion
    strictOutOfSync: false,
```